src/lib.rs

```diff
@@ -98,27 +99,62 @@ impl Puzzle {
         (acrosses, downs)
     }
 
-    pub fn from_str(s: &str) -> Puzzle {
+    /// Construct a Puzzle from a string view
+    ///
+    /// #Note About Grapheme Clusters
+    /// [grapheme clusters](https://www.unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries)
+    /// are a subtle aspect of unicode.
+    /// - in short, a "user-perceived character" may correspond to a cluster
+    ///   of one or more unicode characters.
+    /// - As I understand it, you can think of most of these characters as
+    ///   "modifiers" (I believe a g with grave-accent is a "g" followed by a
+    ///   grave-modifier character). BE AWARE: This mental model may not apply
+    ///   for some characters used to represent non-latin-alphabet languages.
+    /// - In any case, a grapheme cluster is an approximation for these
+    ///   clusters of letters
+    ///
+    /// In the future, a crossword puzzle should fully support arbitrary
+    /// grapheme clusters. For now, this constructor, will parse the cluster,
+    /// and report an error.
+    pub fn from_str(s: &str) -> Result<Puzzle, &'static str> {
         let v: Vec<&str> = s.split('\n').collect();
-        let ncols = v[0].len();
+        let ncols = v[0].graphemes(true).count();
         let nrows = v.len();
         let mut grid = Array::from_elem((nrows, ncols), None);
 
         for i in 0..nrows {
-            for (j, chr) in v[i].char_indices() {
-                grid[[i, j]] = match chr {
-                    '.' => None,
-                    other => Some(other),
+            let mut j = 0;
+            let mut it = UnicodeSegmentation::graphemes(v[i], true);
+            while let Some(grapheme) = it.next() {
```

Comment:
Can this be done without an explicit iterator, e.g. some kind of for loop?

Reply:
I've taken the liberty of changing this.  Feel free to dispute if you don't like the change or you think it's wrong.

Reply:
No idea why I didn't write it this way. The for-loop you wrote looks much better!